### PR TITLE
add job register preservecounts param, update job versions response

### PR DESCRIPTION
**Pull request by drewbailey**
- Document job register payload
- Update job versions job output

### CODE REVIEW

**Review by DingoEatingFuzz — APPROVED** [review]
Wow, the job versions response is totally different than what was documented 😳

It would be nice to also get the secret `diffs` query param for job versions documented, but this is good to merge even without it.

**Comment by drewbailey** [comment]
@DingoEatingFuzz I did not know that query param existed, added, not entirely confident of the wording though.

File: website/pages/api-docs/jobs.mdx [hunk 1]
```diff
@@ -558,6 +565,9 @@ The table below shows this endpoint's support for
 
 ### Parameters
 
+- `diffs` `(bool: false)` - Specifies if the Diffs field should be populated,
+  containing the difference between the current and last job version.
```
Comment (DingoEatingFuzz) [inline website/pages/api-docs/jobs.mdx:569]: Maybe something like "containing the structured diff between the current and last job"? Something to make it clear that the diff here is a JSON object from which a pretty-printed diff like seen in the CLI can be constructed.

Comment (drewbailey) [reply website/pages/api-docs/jobs.mdx:569]: I think the tricky bit is that the returned `Diffs` are an array, that must be inferred by index which version it is a diff of :/ Your suggestion is a good improvement, its definitely a tricky response

**Pull request merged**